openshift.py

```diff
@@ -0,0 +1,750 @@
+import kubernetes
+import pprint
+import logging
+import requests
+import json
+import re
+from flask import Flask, redirect, url_for, request, Response
+
+import sys
+
+# application = Flask(__name__)
+
+
+class openshift:
+    headers = None
+    verify = False
+    url = None
+
+    def __init__(self, url, token, logger):
+        self.set_token(token)
+        self.set_url(url)
+        self.logger = logger
+
+    def set_token(self, token):
+        self.headers = {
+            "Authorization": "Bearer " + token,
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+        }
+
+    def set_url(self, url):
+        self.url = url
+
+    def get_url(self):
+        return self.url
+
+    def cnvt_project_name(self, project_name):
+        suggested_project_name = re.sub("^[^A-Za-z0-9]+", "", project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9]+$", "", suggested_project_name)
+        suggested_project_name = re.sub("[^A-Za-z0-9\-]+", "-", suggested_project_name)
```

Comment:
That `\-` here is an invalid escape sequence. You don't need to escape the `-` at the end of the character class. You can just write `[^A-Za-z0-9-]+`.